### PR TITLE
Implement policy for script command and package dependencies and build/clean scripts

### PR DESCRIPTION
**Pull request by curtisman**
Port a couple of policy checks from fluid-build:
- package with `build` script should have a `clean` script
- Tools used in script commands should be in the dependencies (including fixes for detected issues).  The mapping of command to packages is configurable in `fluidBuildConfig.cjs`

Also implemented a per handler exclusion, need by some package that `build` script doesn't really produce any output, so `clean` script is not necessary.

Misc changes:
- Deleted the ported policy check from `fluid-build`
- Clean up unused code code to execute `clean` in `fluid-build` (which is now alias of -t clean)
- Clean up unused config `requiresTilde` in `fluidBuildConfig.cjs`
- Clean up extraneous scripts in the root `build-tools` release group that was copied over from other places.

### CODE REVIEW

File: build-tools/package.json [hunk 1]
```diff
@@ -42,28 +40,17 @@
...
-		"test:jest:report": "assign-test-ports && lerna run test:jest --concurrency 4 --stream --no-bail --no-sort -- -- --ci --reporters=default --reporters=jest-junit",
 		"test:mocha": "lerna run test:mocha --stream --no-bail --no-sort",
 		"test:mocha:bail": "lerna run test:mocha --stream",
 		"test:mocha:report": "lerna run test:mocha --stream --no-bail --no-sort -- -- --reporter xunit --reporter-option output=nyc/mocha-junit-report.xml",
-		"test:realsvc": "lerna run test:realsvc --stream --no-bail --no-sort",
-		"test:realsvc:report": "lerna run test:realsvc:report --stream --no-bail --no-sort",
-		"test:report": "npm run test:mocha:report && npm run test:jest:report && npm run test:realsvc:report",
+		"test:report": "npm run test:mocha:report",
 		"tsc": "lerna run tsc --stream",
 		"tsc:fast": "fluid-build --root .. -t tsc --buildTools",
```
Comment (tylerbutler) [inline build-tools/package.json:52]: Might as well replace this with the new arguments.

File: tools/api-markdown-documenter/package.json [hunk 2]
```diff
@@ -78,6 +78,7 @@
 		"chai": "^4.3.7",
 		"concurrently": "^8.0.1",
 		"copyfiles": "^2.4.1",
+		"cross-env": "^7.0.3",
```
Comment (tylerbutler) [inline tools/api-markdown-documenter/package.json:81]: Where is this used?

File: fluidBuild.config.cjs [hunk 3]
```diff
@@ -119,20 +119,28 @@ module.exports = {
 			"tools/markdown-magic/test",
 			"tools/telemetry-generator/package-lock.json", // Workaround to allow version 2 while we move it to pnpm
 		],
+		// Exclusion per handler
+		handlerExclusions: {
+			"npm-package-json-script-clean": [
```
Comment (tylerbutler) [inline fluidBuild.config.cjs:124]: Add a comment with the info from the pr description. That is, the part that explains why these packages don't have clean scripts.

File: fluidBuild.config.cjs [hunk 4]
```diff
@@ -119,20 +119,28 @@ module.exports = {
...
-				"eslint-config-prettier",
-				"eslint-plugin-eslint-comments",
-				"eslint-plugin-import",
-				"eslint-plugin-unicorn",
-				"eslint-plugin-unused-imports",
-				"eslint",
-				"prettier",
-				"typescript",
-				"webpack-dev-server",
+			// use by npm-package-json-script-dep policy
```
Comment (tylerbutler) [inline fluidBuild.config.cjs:130]: Is this the only place it's used?

File: fluidBuild.config.cjs [hunk 5]
```diff
@@ -119,20 +119,28 @@ module.exports = {
...
+			// A list of script command and the packages it comes from
```
Comment (tylerbutler) [inline fluidBuild.config.cjs:131]:
```suggestion
			// A list of script commands and the package that contains the command
```

**Review by tylerbutler — APPROVED** [review]

File: fluidBuild.config.cjs (hunk 4 above)
Comment (curtisman) [reply fluidBuild.config.cjs:130]: yes

File: build-tools/package.json (hunk 1 above)
Comment (curtisman) [reply build-tools/package.json:52]: Done.

File: tools/api-markdown-documenter/package.json (hunk 2 above)
Comment (curtisman) [reply tools/api-markdown-documenter/package.json:81]: The script `test:mocha:multireport`

**Pull request merged**